magic-nix-cache/src/flakehub.rs

```diff
@@ -137,24 +98,39 @@ pub async fn init_cache(
 
     // Periodically refresh JWT in GitHub Actions environment
     if environment.is_github_actions() {
-        if let super::FlakeHubAuthSource::Netrc(path) = auth_method {
-            // NOTE(cole-h): This is a workaround -- at the time of writing, GitHub Actions JWTs are only
-            // valid for 5 minutes after being issued. FlakeHub uses these JWTs for authentication, which
-            // means that after those 5 minutes have passed and the token is expired, FlakeHub (and by
-            // extension FlakeHub Cache) will no longer allow requests using this token. However, GitHub
-            // gives us a way to repeatedly request new tokens, so we utilize that and refresh the token
-            // every 2 minutes (less than half of the lifetime of the token).
-            let netrc_path_clone = path.to_path_buf();
-            let initial_github_jwt_clone = flakehub_password.clone();
-            let flakehub_cache_server_clone = flakehub_cache_server.to_string();
-            let api_clone = api.clone();
-
-            tokio::task::spawn(refresh_github_actions_jwt_worker(
-                netrc_path_clone,
-                initial_github_jwt_clone,
-                flakehub_cache_server_clone,
-                api_clone,
-            ));
+        match auth_method {
+            super::FlakeHubAuthSource::Netrc(path) => {
+                let netrc_path_clone = path.to_path_buf();
+                let initial_github_jwt_clone = flakehub_password.clone();
+                let flakehub_cache_server_clone = flakehub_cache_server.to_string();
+                let api_clone = api.clone();
+
+                tokio::task::spawn(refresh_github_actions_jwt_worker(
+                    netrc_path_clone,
+                    initial_github_jwt_clone,
+                    flakehub_cache_server_clone,
+                    api_clone,
+                ));
+            }
+            crate::FlakeHubAuthSource::DeterminateNixd => {
+                let api_clone = api.clone();
+                let netrc_file = PathBuf::from(DETERMINATE_NETRC_PATH);
+                let flakehub_api_server_clone = flakehub_api_server.clone();
+                let flakehub_cache_server_clone = flakehub_cache_server.clone();
+
+                let initial_meta = tokio::fs::metadata(&netrc_file).await.map_err(|e| {
+                    Error::Io(e, format!("getting metadata of {}", netrc_file.display()))
+                })?;
+                let initial_inode = initial_meta.ino();
+
+                tokio::task::spawn(refresh_determinate_token_worker(
+                    netrc_file,
+                    initial_inode,
+                    flakehub_api_server_clone,
+                    flakehub_cache_server_clone,
+                    api_clone,
+                ));
+            }
```

Comment:
nit: maybe move this out to unconditionally enable for when dnixd is the auth source.